pyccel/codegen/printing/fcode.py

```diff
@@ -2663,6 +2666,30 @@ def _print_NumpySign(self, expr):
         func = PyccelFunctionDef('numpy_sign', NumpySign)
         self._additional_imports.add(Import('numpy_f90', AsName(func, 'numpy_sign')))
         return f'numpy_sign({self._print(expr.args[0])})'
+    
+    def _print_NumpyIsNan(self, expr):
+        """
+        Convert a Python expression with a numpy isnan function call to Fortran function call
+        """
+        arg = expr.args[0]
+        code_arg = self._print(arg)
+        return f"isnan({code_arg})"
+
+    def _print_NumpyIsInf(self, expr):
+        """
+        convert a python expression with a numpy isinf function call to Fortran function call
+        """
+        arg = expr.args[0]
+        code_arg = self._print(arg)
+        return f"isinf({code_arg})"
+    
+    def _print_NumpyIsFinite(self,expr):
+        """
+        conver a pytho expression with a numpy isnan function call to Fortran function call
+        """
+        arg = expr.args[0]
+        code_arg = self._print(arg)
+        return f"isfinite({code_arg})"
```

Comment:
These functions aren't using the values that you added to `numpy_ufunc_to_fortran`. It seems that the functions that you are trying to use are not recognised, so I think it would be better to use the functions that are defined in the `ieee_arithmetic` intrinsic module. 

To use these objects you will also need to make sure that they are correctly imported. This can be done with:
```python
self._constantImports.setdefault('ieee_arithmetic', set()).add('ieee_is_finite')
```